honeybee/model.py

```diff
@@ -22,76 +25,235 @@ class Model(object):
 
     Properties:
         name
-        name_original
+        display_name
         north_angle
         north_vector
         rooms
         faces
         shades
         apertures
         doors
-        has_ngon_apertures
-        had_ngon_doors
+        orphaned_faces
+        orphaned_shades
+        orphaned_apertures
+        orphaned_doors
     """
-    __slots__ = ('_name', '_name_original', '_rooms', '_faces', '_shades',
-                 '_apertures', '_doors', '_room_names', '_face_names', '_shade_names',
-                 '_aperture_names', '_door_names', '_properties')
+    __slots__ = ('_name', '_display_name', '_rooms', '_faces', '_shades',
+                 '_apertures', '_doors', '_properties', '_north_angle', '_north_vector')
 
-    def __init__(self, name, objects, north_angle=0):
+    def __init__(self, name, rooms=[], orphaned_faces=[], orphaned_shades=[],
```

Comment:
default values should not be set to empty lists.

Reminder here: https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

Reply:
Ok. I will add an extra check for `None`, set defaults to `None`, and still permit the setting of empty lists, since that is what will be the case for most models.